### PR TITLE
feat(droid): Enable Textbox.CharacterCasing

### LINKED ISSUE #1574: [Droid]TextBox.CharacterCasing is not respected
## Current behavior
If I set the Property CharacterCasing on TextBox via XAML (e.g. to "Lower"), it works on UWP but not on Android. Did not test WASM and iOs.

## Expected behavior
The CharacterCasing is respected on Android, too.

## How to reproduce it (as minimally and precisely as possible)
Add a TextBox to a page and set it's CharacterCasing to "Lower". On UWP you can't enter uppercase Characters. On Android, the upperace characters are shown.

## Environment

Nuget Package: v1.46.230-dev.2733

Package Version(s):

Affected platform(s):
- [ ] iOS
- [X] Android
- [ ] WebAssembly
- [ ] Windows
- [ ] Build tasks

Visual Studio
- [ ] 2017 (version: )
- [X] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins
- [ ] Resharper (version: )

## Anything else we need to know?
See Remarks [here](https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.textbox.charactercasing?view=netframework-4.8). Maybe the characters are entered programmatically on Android?

### PR DESCRIPTION
**Pull request by ajpinedam**
GitHub Issue (If applicable): #
closes #1574 

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?
The CharacterCasing was not implemented 

## What is the new behavior?
The CharacterCasing is implemented. A TextBox will follow the CharacterCasing (Upper, Lower, Normal) indicated.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by MartinZikmund** [comment]
@ajpinedam I haven't checked, but could this be implemented without platform-specific code by utilizing the path that `TextBox.TextChanging` uses? I think it worked fine when I used it on one of my projects

**Comment by jeromelaban** [comment]
Thanks Andres, can you add a UI Test ?

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.iOS.cs [hunk 1]
```diff
@@ -324,5 +324,20 @@ private void OnKeyboardAppearanceChanged(UIKeyboardAppearance newValue)
...
+			var casing = (CharacterCasing)e.NewValue;
+			switch (casing)
+			{
+				case CharacterCasing.Lower:
+
+					break;
+				case CharacterCasing.Upper:
+					break;
+				case CharacterCasing.Normal:
+					break;
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.iOS.cs:339]: Should it be kept here ?

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs [hunk 2]
```diff
@@ -435,6 +436,29 @@ private void OnTextWrappingChanged(DependencyPropertyChangedEventArgs args)
 
 		#endregion
 
+		public CharacterCasing CharacterCasing
+		{
+			get => (CharacterCasing)this.GetValue(CharacterCasingProperty);
+			set => this.SetValue(CharacterCasingProperty, value);
+		}
+
+		public static DependencyProperty CharacterCasingProperty { get; } =
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs:445]: We should still keep the NotImplemented attribute for this and the property above for platforms that do not support it.

**Comment by ajpinedam** [comment]
@jeromelaban both comments were resolved

**Comment by jeromelaban** [comment]
Thanks andres! Could you squash your commits to it's only in one ? Thank!

**Review by jeromelaban — APPROVED** [review]

**Comment by ajpinedam** [comment]
\azp run

**Pull request merged**